firebase-firestore/src/main/java/com/google/firebase/firestore/remote/FirebaseClientGrpcMetadataProvider.java

```diff
@@ -45,8 +45,12 @@ public DefaultGrpcMetadata(
   @Override
   public void updateMetadata(@NonNull Metadata metadata) {
     metadata.put(USER_AGENT_HEADER, userAgentPublisher.getUserAgent());
-    metadata.put(
-        HEART_BEAT_HEADER,
-        Integer.toString(heartBeatInfo.getHeartBeatCode(firebaseFirestoreHeartBeatTag).getCode()));
+    int heartBeatCode = heartBeatInfo.getHeartBeatCode(firebaseFirestoreHeartBeatTag).getCode();
+    if (heartBeatCode != 0) {
```

Comment:
Since we can't link to the design doc for this here, could you add a comment about why `0` is special here?

```
// Non-zero values indicate some kind of heartbeat should be sent
```

Reply:
yes sure! Thanks!